internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,444 @@
+	output.Println("Adding tags at schema level and topic level.")
+	if _, _, err := addSchemaTags(details, spec.Components, topicName, schemaId); err != nil {
+		return err
+	}
+	if !topicNotCreated && spec.Channels != nil {
+		if _, _, err := addTopicTags(details, spec.Channels[topicName].Subscribe, topicName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createExportFlagsFromImportFlags(flagsImp *flagsImport) flags {
+	flagsExport := flags{
+		kafkaApiKey:             flagsImp.kafkaApiKey,
+		schemaRegistryApiKey:    flagsImp.schemaRegistryApiKey,
+		schemaRegistryApiSecret: flagsImp.schemaRegistryApiSecret,
+	}
+	return flagsExport
+}
+
+func isModifiableConfig(configName string) bool {
+	modifiableConfigs := []string{"delete.retention.ms", "max.compaction.lag.ms", "max.message.bytes", "message.timestamp.difference.max.ms", "message.timestamp.type",
+		"min.compaction.lag.ms", "min.insync.replicas", "retention.bytes", "retention.ms", "segment.bytes", "segment.ms"}
+	for _, config := range modifiableConfigs {
+		if config == configName {
+			return true
+		}
+	}
+	return false
+}
+
+func (c *command) addTopic(details *accountDetails, topicName string, description string, kb KafkaBinding, overwrite bool) (bool, error) {
+	topicConfigs := []kafkarestv3.CreateTopicRequestDataConfigs{}
+	updateConfigs := []kafkarestv3.AlterConfigBatchRequestDataData{}
+	for configName, configValue := range kb.XConfigs {
+		value := configValue
+		topicConfigs = append(topicConfigs, kafkarestv3.CreateTopicRequestDataConfigs{
+			Name:  configName,
+			Value: *kafkarestv3.NewNullableString(&value),
+		})
+		if isModifiableConfig(configName) {
+			updateConfigs = append(updateConfigs, kafkarestv3.AlterConfigBatchRequestDataData{
+				Name:  configName,
+				Value: *kafkarestv3.NewNullableString(&value),
+			})
+		}
+	}
+	topicRequestData := kafkarestv3.CreateTopicRequestData{
+		TopicName: topicName,
+		Configs:   &topicConfigs,
+	}
+	// Check if topic already exists
+	for _, topics := range details.topics {
+		if topics.TopicName == topicName {
+			// Topic already exists
+			log.CliLogger.Info("Topic is already present.")
+			if !overwrite {
+				// Do not overwrite existing topic. Move to the next topic.
+				return true, nil
+			}
+			// Overwrite topic configs
+			log.CliLogger.Info("Overwriting topic configs")
+			_, err := c.kafkaRest.CloudClient.UpdateKafkaTopicConfigBatch(details.clusterId, topicName, kafkarestv3.AlterConfigBatchRequestData{Data: updateConfigs})
+			if err != nil {
+				return true, fmt.Errorf("unable to update topic config delete.retention.ms: %v", err)
+			}
+			if description != "" {
+				err = addTopicDescription(details.srClient, details.srContext, details.clusterId+":"+topicName, description)
+			}
+			return true, err
+		}
+	}
+	log.CliLogger.Infof("Topic not found. Adding a new topic: %s", topicName)
+	_, httpResp, err := c.kafkaRest.CloudClient.CreateKafkaTopic(details.clusterId, topicRequestData)
+	if err != nil {
```

Comment:
It's good practice to limit scope where possible.
```suggestion
	if _, httpResp, err := c.kafkaRest.CloudClient.CreateKafkaTopic(details.clusterId, topicRequestData); err != nil {
```